### PR TITLE
refactor(components): use `useResizeObserver`

### PR DESCRIPTION
**Pull request by CarterLi**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

`addResizeListener` and `removeResizeListener` are deprecated. Use `useResizeObserver` in `vueuse` as suggested

### CODE REVIEW

File: packages/components/cascader/src/index.vue [hunk 1]
```diff
@@ -723,11 +720,7 @@ export default defineComponent({
         inputEl?.offsetHeight ||
         INPUT_HEIGHT_MAP[realSize.value] ||
         DEFAULT_INPUT_HEIGHT
-      addResizeListener(inputEl, updateStyle)
-    })
-
-    onBeforeUnmount(() => {
-      removeResizeListener(input.value?.$el, updateStyle)
+      useResizeObserver(inputEl, updateStyle)
```
Comment (sxzz) [inline packages/components/cascader/src/index.vue:723]: Move it outside of `onMounted`

File: packages/components/select-v2/src/useSelect.ts [hunk 2]
```diff
@@ -750,11 +738,7 @@ const useSelect = (props: ExtractPropTypes<typeof SelectProps>, emit) => {
 
   onMounted(() => {
     initStates()
-    addResizeListener(selectRef.value, handleResize)
-  })
-
-  onBeforeMount(() => {
-    removeResizeListener(selectRef.value, handleResize)
+    useResizeObserver(selectRef, handleResize)
```
Comment (sxzz) [inline packages/components/select-v2/src/useSelect.ts:741]: Same here

File: packages/components/cascader/src/index.vue (hunk 1 above)
Comment (CarterLi) [reply packages/components/cascader/src/index.vue:723]: `inputEl` aka `input.value?.$el` is not a ref, so I don't think it's possible

**Comment by CarterLi** [comment]
Done @sxzz

**Review by sxzz — APPROVED** [review]

**Pull request merged**